paho/client.go

```diff
@@ -22,29 +22,36 @@ type (
 	// are required to be set, defaults are provided for Persistence, MIDs,
 	// PacketTimeout and Router.
 	ClientConfig struct {
-		ClientID      string
 		Conn          net.Conn
 		MIDs          MIDService
 		AuthHandler   Auther
 		Router        Router
 		Persistence   Persistence
 		PacketTimeout time.Duration
 		OnDisconnect  func(*Disconnect)
+		OnClose       func()
 	}
 	// Client is the struct representing an MQTT client
 	Client struct {
 		ClientConfig
-		// caCtx is used for synchronously handling the connect/connack
-		// flow, raCtx is used for handling the MQTTv5 authentication
-		// exchange.
+		// caCtx is used for synchronously handling the connect/connack flow
+		// raCtx is used for handling the MQTTv5 authentication exchange.
+
+		connectOnce sync.Once
+		ca          *Connack
+		cerr        error
```

Comment:
Would you mind adding a comment on this one? On `ca` at least we got the type but I think `cerr` could use a comment (or just a longer name).

Reply:
Added.